lib/bundler_api/cdn.rb

```diff
@@ -3,17 +3,21 @@
 class BundlerApi::Cdn
   def self.purge_specs(client = self.client)
     return unless client
-    client.purge_key  'dependencies'
-    client.purge_path '/latest_specs.4.8.gz'
-    client.purge_path '/specs.4.8.gz'
-    client.purge_path '/prerelease_specs.4.8.gz'
+    threads = []
+    threads << client.purge_key('dependencies')
+    threads << client.purge_path( '/latest_specs.4.8.gz')
+    threads << client.purge_path( '/specs.4.8.gz')
+    threads << client.purge_path( '/prerelease_specs.4.8.gz')
+    threads.each { |t| t.join }
```

Comment:
That's better, although I would create the thread at this level and not in `purge_key` or `purge_path`.

Something like:

```ruby
threads = []
threads << Thread.new { client.purge_key('dependencies') }
threads << Thread.new { client.purge_path( '/latest_specs.4.8.gz') }
# ...
threads.each(&:join)
```